projects/canopy/src/lib/spacing/padding/padding.directive.ts

```diff
@@ -16,62 +18,73 @@ import { ResponsiveSpacing, SpacingVariant } from '../spacing.interface';
     this.paddingTopClasses = this.toggleClasses(newClasses, this.paddingTopClasses);
   }
 
   paddingRightClasses: Array<string> = [];
   @Input()
-  set lgPaddingRight(padding: SpacingVariant | ResponsiveSpacing) {
+  set lgPaddingRight(padding: SpacingVariant | SpacingResponsive) {
     const newClasses = this.createNewClasses(padding, 'lg-padding__right');
     this.paddingRightClasses = this.toggleClasses(newClasses, this.paddingRightClasses);
   }
 
   paddingBottomClasses: Array<string> = [];
   @Input()
-  set lgPaddingBottom(padding: SpacingVariant | ResponsiveSpacing) {
+  set lgPaddingBottom(padding: SpacingVariant | SpacingResponsive) {
     const newClasses = this.createNewClasses(padding, 'lg-padding__bottom');
     this.paddingBottomClasses = this.toggleClasses(newClasses, this.paddingBottomClasses);
   }
 
   paddingLeftClasses: Array<string> = [];
   @Input()
-  set lgPaddingLeft(padding: SpacingVariant | ResponsiveSpacing) {
+  set lgPaddingLeft(padding: SpacingVariant | SpacingResponsive) {
     const newClasses = this.createNewClasses(padding, 'lg-padding__left');
     this.paddingLeftClasses = this.toggleClasses(newClasses, this.paddingLeftClasses);
   }
 
   @Input()
-  set lgPaddingHorizontal(padding: SpacingVariant | ResponsiveSpacing) {
+  set lgPaddingHorizontal(padding: SpacingVariant | SpacingResponsive) {
     this.lgPaddingLeft = padding;
     this.lgPaddingRight = padding;
   }
 
   @Input()
-  set lgPaddingVertical(padding: SpacingVariant | ResponsiveSpacing) {
+  set lgPaddingVertical(padding: SpacingVariant | SpacingResponsive) {
     this.lgPaddingTop = padding;
     this.lgPaddingBottom = padding;
   }
 
   paddingClasses: Array<string> = [];
   @Input()
-  set lgPadding(padding: SpacingVariant | ResponsiveSpacing) {
-    const newClasses = this.createNewClasses(padding, 'lg-padding');
+  set lgPadding(padding: SpacingVariant | SpacingResponsive) {
+    const newClasses = this.createNewClasses(padding, 'padding');
     this.paddingClasses = this.toggleClasses(newClasses, this.paddingClasses);
   }
 
-  createNewClasses(padding: SpacingVariant | ResponsiveSpacing, classPrefix: string) {
+  createNewClasses(spacing: SpacingVariant | SpacingResponsive, spacingProp: string) {
     const newClasses = [];
-    if (!padding) {
+    if (!spacing) {
       return [];
     }
-    if (typeof padding === 'object') {
-      Object.keys(padding).forEach((key) => {
-        newClasses.push(`${classPrefix}--${key}--${padding[key]}`);
+    if (typeof spacing === 'object') {
+      Object.keys(spacing).forEach((key) => {
+        const className = `lg-${spacingProp}--${key}--${spacing[key]}`;
+        const spacingKey = spacing[key];
+        const styleRule = `${spacingProp}: ${SpacingValues[spacingKey]}`;
+        this.dynamicStyleService.addStyleToMediaQuery(
+          className,
+          styleRule,
+          BreakpointValues[key],
+        );
+        newClasses.push(className);
```

Comment:
Tempted to ditch the assignments for brevity, but that could be a style opinion.
```
this.dynamicStyleService.addStyleToMediaQuery(
  `lg-${spacingProp}--${key}--${spacing[key]}`,
   `${spacingProp}: ${SpacingValues[spacing[key]]},
   BreakpointValues[key],
);
```

Reply:
This has changed quite a bit and is now in the **spacing.service.ts** file